api/v1/_uploadPublicFile/PKPUploadPublicFileHandler.inc.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * @file api/v1/contexts/PKPUploadPublicFileHandler.inc.php
+ *
+ * Copyright (c) 2014-2019 Simon Fraser University
+ * Copyright (c) 2000-2019 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class PKPUploadPublicFileHandler
+ * @ingroup api_v1_uploadPublicFile
+ *
+ * @brief Handle API requests to upload a file to a user's public directory.
+ */
+import('lib.pkp.classes.handler.APIHandler');
+
+class PKPUploadPublicFileHandler extends APIHandler {
+	/**
+	 * @copydoc APIHandler::__construct()
+	 */
+	public function __construct() {
+		$this->_handlerPath = '_uploadPublicFile';
+		$roles = [ROLE_ID_SITE_ADMIN, ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR, ROLE_ID_REVIEWER, ROLE_ID_AUTHOR, ROLE_ID_ASSISTANT, ROLE_ID_READER];
+		$this->_endpoints = array(
+			'OPTIONS' => array(
+				array(
+					'pattern' => $this->getEndpointPattern(),
+					'handler' => array($this, 'getOptions'),
+					'roles' => $roles,
+				),
+			),
+			'POST' => array(
+				array(
+					'pattern' => $this->getEndpointPattern(),
+					'handler' => array($this, 'uploadFile'),
+					'roles' => $roles,
+				),
+			),
+		);
+
+		parent::__construct();
+	}
+
+	/**
+	 * @copydoc PKPHandler::authorize
+	 */
+	public function authorize($request, &$args, $roleAssignments) {
+		import('lib.pkp.classes.security.authorization.PolicySet');
+		$rolePolicy = new PolicySet(COMBINING_PERMIT_OVERRIDES);
+
+		import('lib.pkp.classes.security.authorization.RoleBasedHandlerOperationPolicy');
+		foreach($roleAssignments as $role => $operations) {
+			$rolePolicy->addPolicy(new RoleBasedHandlerOperationPolicy($request, $role, $operations));
+		}
+		$this->addPolicy($rolePolicy);
+
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+
+	/**
+	 * A helper method which adds the necessary response headers to allow
+	 * file uploads
+	 *
+	 * @param $response Response object
+	 * @return Response
+	 */
+	private function getResponse($response) {
+		return $response->withHeader('Access-Control-Allow-Headers', 'Content-Type, X-Requested-With, X-PINGOTHER, X-File-Name, Cache-Control');
+	}
+
+	/**
+	 * Upload a requested file
+	 *
+	 * @param $slimRequest Request Slim request object
+	 * @param $response Response object
+	 * @param $args array arguments
+	 * @return Response
+	 */
+	public function uploadFile($slimRequest, $response, $args) {
+		$request = $this->getRequest();
+
+		if (empty($_FILES) || empty($_FILES['file'])) {
+			return $response->withStatus(400)->withJsonError('api.temporaryFiles.400.noUpload');
+		}
+
+		$siteDir = Core::getBaseDir() . '/' . Config::getVar('files', 'public_files_dir') . '/site';
+
+		if (!file_exists($siteDir) || !is_writeable($siteDir)) {
+			return $response->withStatus(500)->withJsonError('api.publicFiles.500.badFilesDir');
+		}
+		$userDir = $siteDir . '/images/' . $request->getUser()->getUsername();
+		$isUserAllowed = true;
+		$allowedDirSize = 25000 * 1024; // 25mb
```

Comment:
This default is worth adding to `config.TEMPLATE.inc.php` -- one quick way to disable file uploads entirely, without resorting to a plugin, would be to set it to zero. (25MB could continue to be a default if the `config.inc.php` value isn't specified -- see `Config::getVar` on defaults. But I wonder if something much smaller wouldn't be better, i.e. 5MB...)

Reply:
Good idea!

Reply:
https://github.com/pkp/pkp-lib/commit/f86f3f89be82b0d1d3c54e3aefb7307fb66db31d, https://github.com/pkp/ojs/commit/dfeb0ad5768b8a876b7cdabf15eff692d11c5030 and https://github.com/pkp/omp/commit/a177ed0dcebae5d04c5d46baebca82784e1433f9